core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -421,6 +421,10 @@ public void onComplete(Map<TopicIdPartition, Set<Long>> offsetsMap, Exception ex
                     Set<Long> mergedOffsets = new HashSet<>();
                     mergedOffsets.addAll(oldOffsets);
                     mergedOffsets.addAll(newOffsets);
+                    if (mergedOffsets.size() < (oldOffsets.size() + newOffsets.size())) {
```

Comment:
I think this is quite a dangerous thing to leave in here. In principle, a test could rely upon repeated delivery of a record (timeout for example) which could actually call the callback twice. I'd remove this because I predict it will be the source of a flaky test later on.

Reply:
Ah right makes sense, sometimes genuinely we can have the callback called twice. I have removed it now, thanks.